ci-operator/jobs/openshift/openshift-ansible/openshift-openshift-ansible-presubmits.yaml

```diff
@@ -308,6 +308,197 @@ presubmits:
+              name: ci-operator-openshift-openshift-ansible
+              key: release-3.10.json
+        - name: RPM_REPO_BASEURL_REF
+          value: https://storage.googleapis.com/origin-ci-test/releases/openshift/origin/release-3.10/.latest-rpms
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+          set -e
+          export RPM_REPO="$( curl -q "${RPM_REPO_BASEURL_REF}" 2>/dev/null)"
+          export IMAGE_UPGRADE_ANSIBLE=${IMAGE_ANSIBLE}
+          export UPGRADE_RPM_REPO=${RPM_REPO}
+          ci-operator \
+            --artifact-dir=$(ARTIFACTS) \
+            --secret-dir=/usr/local/e2e-gcp-cluster-profile --template=/usr/local/e2e-gcp \
+            --target=e2e-gcp
+
+  - name: pull-ci-openshift-openshift-ansible-e2e-gcp-upgrade-311-to-311x
+    agent: kubernetes
+    context: ci/prow/gcp-minor-upgrade
+    branches:
+    - master
+    rerun_command: "/test gcp-minor-upgrade"
+    always_run: true
+    trigger: "((?m)^/test( all| gcp-minor-upgrade),?(\\s+|$))"
+    decorate: true
+    skip_cloning: true
+    spec:
+      serviceAccountName: ci-operator
+      volumes:
+      - name: job-definition
+        configMap:
+          name: prow-job-cluster-launch-e2e-upgrade
+      - name: cluster-profile
+        projected:
+          sources:
+          - secret:
+              name: cluster-secrets-gcp
+          - configMap:
+              name: cluster-profile-gcp
+      containers:
+      - name: test
+        image: ci-operator:latest
+        volumeMounts:
+        - name: job-definition
+          mountPath: /usr/local/e2e-gcp
+          subPath: cluster-launch-e2e-upgrade.yaml
+        - name: cluster-profile
+          mountPath: /usr/local/e2e-gcp-cluster-profile
+        env:
+        - name: TEST_FOCUS
+          value: Suite:openshift/conformance/parallel
+        - name: JOB_NAME_SAFE
+          value: e2e-gcp
+        - name: CLUSTER_TYPE
+          value: gcp
+        - name: CONFIG_SPEC
+          valueFrom:
+            configMapKeyRef:
+              name: ci-operator-openshift-openshift-ansible
+              key: master.json
+        - name: RPM_REPO_BASEURL_REF
+          value: https://storage.googleapis.com/origin-ci-test/releases/openshift/origin/master/.latest-rpms
+        - name: TEST_UPGRADE
+          value: "v3_11"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+          set -e
+          export RPM_REPO="$( curl -q "${RPM_REPO_BASEURL_REF}" 2>/dev/null)"
+          export IMAGE_UPGRADE_ANSIBLE=${IMAGE_ANSIBLE}
+          export UPGRADE_RPM_REPO=${RPM_REPO}
+          ci-operator \
+            --artifact-dir=$(ARTIFACTS) \
+            --secret-dir=/usr/local/e2e-gcp-cluster-profile --template=/usr/local/e2e-gcp \
+            --target=e2e-gcp
+
+  - name: pull-ci-openshift-openshift-ansible-e2e-gcp-upgrade-310-to-311
+    agent: kubernetes
+    context: ci/prow/gcp-major-upgrade
+    branches:
+    - release-3.10
```

Comment:
Shouldn't this run against 3.11 and use the 3.11 config?

Reply:
This is major upgrade so we start with 3.10, mount 3.11 image (`IMAGE_UPGRADE_ANSIBLE`), update `openshift_additional_repos` with `UPGRADE_RPM_REPO` and run the upgrade playbook there.

This however would use 3.11 tests - are these incompatible?
Perhaps we should do the other way around - have `IMAGE_PREVIOUS_ANSIBLE` and `PREVIOUS_RPM_REPO` for setup, run upgrade with `IMAGE_ANSIBLE` and have tests from `IMAGE_TESTS`

Reply:
Reworked this with var renamed